perf: removed unused code with a 100% confidence from vulture

mehedikhan72 opened the pull request:
#32603 

This commit removed unused code(specifically unused variables, imports and one instance of code which could not run logically) that were shown with a 100% confidence by [vulture](https://adamj.eu/tech/2023/07/12/django-clean-up-unused-code-vulture/), running this [script](https://github.com/openedx/edx-platform/blob/master/scripts/vulture/find-dead-code.sh).

There were many other unused variables for example, but I left them untouched and deleted only the ones shown by vulture.

Commit:
feat: BLOCK_STRUCTURES_SETTINGS['PRUNING_ACTIVE'] toggle removed depr31

Yagnesh1998 committed:
feat: Update test_models.py

UsamaSadiq committed:
fix: remove language_session_key usage

iamsobanjaved committed:
fix: update urls and tests for cookies

Yagnesh1998 committed:
feat: Update models.py

Yagnesh1998 committed:
feat: Update models.py 

remove white space.

Commit:
fix: fix mako runtime bug with studio_view. TNL-10912

Yagnesh1998 committed:
feat: Update models.py

Yagnesh1998 committed:
feat: Update models.py

removed white space

Yagnesh1998 committed:
feat: Update models.py

feanil committed:
docs: Add some basic docs on the Rest API.

There is probably a lot more to say but add a quick How-To and concept
docs that we can build on.

mohd-akram committed:
chore: run js tests on all pull requests

mehedikhan72 committed:
perf: removed unused code with a 100% confidence from vulture

openedx-webhooks commented:
Thanks for the pull request, @mehedikhan72!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

mehedikhan72 commented:
@dianakhuang , here is the PR. I need a maintainer's approval to run the checks.

swayamrana committed:
feat: pact provider profile setup

bcitro committed:
chore: upgrade edx-enterprise to 4.0.12

timmc-edx committed:
Revert "feat: Pact Provider Verification for Profile Endpoint" (#32999)

Reverts openedx/edx-platform#32928 -- mistaken merge.

mohd-akram committed:
feat: allow disabling default beta tester enrollment notifications

The "Notify users by email" checkbox checks the value of the 
`BATCH_ENROLLMENT_NOTIFY_USERS_DEFAULT` configuration to match the user 
experience with other checkboxes on the Membership page.

bradenmacdonald committed:
feat: new actions menu for copy/pasting units in Studio (behind waffle flag) (#32891)

Commit:
chore(i18n): update translations

alex-sheehan-edx committed:
feat: Upgrade Python dependency edx-enterprise

4.0.14

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

johnnagro committed:
feat: release edx-enterprise 4.0.13 (#33002)

mehedikhan72 commented:
@dianakhuang , Update on the checks: From what I see, removing unused codes made some issues in the entire codebase. One of them is the functions being called with the old number of parameters. I'm fixing those one by one. Another issue is that some of the unused variables are supposed to be there, at least that is what the errors are saying. I will revert those changes back to normal.

mehedikhan72 committed:
perf: modifed changes to fix check errors 1.0

muhammadadeeltajamul committed:
fix: update discussion enabled if true (#33008)

leangseu-edx committed:
chore: update ora2 package

robrap committed:
feat: add observability for default auth classes (#33003)

Added new authentication classes to be used in
DEFAULT_AUTHENTICATION_CLASSES that include
observability. This will enable us to see more
about the endpoints using the defaults, to help
us make choices about changes in the defaults.

We make the DRF default of Session and Basic
Authentication explicit by setting
DEFAULT_AUTHENTICATION_CLASSES explicitly.

mehedikhan72 committed:
perf: modified changes 2.0

iamsobanjaved committed:
fix: added back test_views file

bradenmacdonald committed:
fix: error on course outline in Studio due to browser caching old JS file (#33015)

This fixes a bug where some users would see an error that prevented the course outline page in Studio from loading, for any course.

timmc-edx committed:
fix: Run `make compile-requirements` to fix inconsistency (#33016)

timmc-edx committed:
build: Fail PR if requirements files are inconsistent (#32057)

We have a few situations where requirements files can become
inconsistent or cause unnecessary review churn in later `make upgrade`
runs either due to manual editing, inconsistent environments, or
incorrectly specified git dependencies. This will produce a failing
check for any PR that does not produce a clean run of `make
compile-requirements` on Linux.

Addresses https://github.com/openedx/edx-platform/issues/31372

UsamaSadiq committed:
fix: fix django4 deprecation warnings (#32948)

UsamaSadiq committed:
fix: fix django4 deprecation warnings (#32957)

awais786 committed:
chore: upgrading `django-simple-history` to latest version. (#33010)

* chore: upgrading `django-simple-history` to latest version.

mumarkhan999 committed:
chore: update pymemcache settings (#33019)

awais786 committed:
feat: replaced django-fernet-fields with django-fernet-fields-v2. (#32990)

* feat: replaced django-fernet-fields with django-fernet-fields-v2.

UsamaSadiq committed:
fix: fix django4 deprecation warnings (#32975)

UsamaSadiq committed:
fix: fix django4 deprecation warnings (#32978)

UsamaSadiq committed:
fix: replace ugettext with gettext (#33022)

AhtishamShahid committed:
fix: removed redundant unit update from task (#33017)

Commit:
chore: geoip2: update maxmind geolite country database

edx-requirements-bot committed:
Python Requirements Update (#33034)

* chore: Updating Python Requirements and revert redis change.

UsamaSadiq committed:
fix: fix is_safe_url and urlquote warnings (#33041)

* fix: fix is_safe_url and urlquote warnings
* fix: replace urlquote with quote

ruzniaievdm committed:
feat: Create DRF for course team (#32782)

feanil committed:
docs: Add more examples related to Auth.

feanil committed:
fixup! docs: Add some basic docs on the Rest API.

feanil committed:
fixup! docs: Add more examples related to Auth.

feanil committed:
fixup! docs: Add some basic docs on the Rest API.

feanil committed:
docs: Make the docs easier to read and and some links.

yusuf-musleh committed:
fix: Prevent multiple dropdowns opening at once (#33046)

This closes already open dropdown menus when opening another dropdown
menu, to prevent them from overlapping each other.

alex-sheehan-edx committed:
feat: Upgrade Python dependency edx-enterprise

restoring old stage historical table

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

UsamaSadiq committed:
fix: fix django.utils.translations warnings (#33040)

* fix: fix ugetext and ngettext deprecation warnings
* fix: fix deprecation warnings from html templates

awais786 committed:
fix: upgrade oauthlib and django-oauth-toolkit to new versions. (#32631)

* chore: bump django-oauth-toolkit and oauthlib
---------

Co-authored-by: Muhammad Umar Khan <m.umarkhan999@gmail.com>

Commit:
docs: Apply suggestions from code review

Typos and other small fixes.

Co-authored-by: Robert Raposa <rraposa@edx.org>

feanil committed:
docs: Clarify comment for getting a refresh token.

yusuf-musleh committed:
feat: Remove component-level copy/paste feature flag (#32980)

This is so that the feature is on by default.

Commit:
docs: Apply suggestions from code review

Commit:
docs: Apply suggestions from code review

Commit:
chore(i18n): update translations

awais786 committed:
chore: Adding test to verify s3boto3 storages with invalid params. (#33058)

* chore: Adding test to verify s3boto3 storages with invalid params.

UsamaSadiq committed:
fix: fix request.is_ajax() deprecation warning (#33055)

* fix: fix request.is_ajax() deprecation warning

syedsajjadkazmii committed:
fix: update cross product recommendations api endpoint docstring (#33066)

VAN-1593

iamsobanjaved committed:
fix: add few deps to base layer to fix reqs installation on dev image (#32985)

bradenmacdonald committed:
fix: refactor MakoService to allow specifying namespace per template (#33061)

germanolleunlp committed:
feat: [ACADEMIC-16209] Unit summary settings (#32855)

* feat: [ACADEMIC-16209] Unit summary settings

[https://jira.2u.com/browse/ACADEMIC-16209](https://jira.2u.com/browse/ACADEMIC-16209)

1. Add unit xpert unit summaries settings behind flag `summaryhook_summaries_configuration` added [here](https://github.com/edx/ai-aside/pull/45/files)
2. Only show the checkbox when the value is a `boolean` otherwise the feature is considerer `disabled` by the flag.
3. Update block handler to update this value via `api` exposed [here](https://github.com/edx/ai-aside/pull/43)
4. Create `AiAsideSummary` configuration class to provide access to the `ai_aside.api` endpoints.

robrap committed:
fix: change session_auth_result value to n/a

Changes one of the values of the custom attribute
session_auth_result from 'skipped' to 'n/a'.

robrap committed:
feat!: remove BasicAuthentication default

Removed BasicAuthentication as a default from DRF
endpoints that have not overridden the authentication
classes. It appears this is not in use, and was just
implicitly a default because it came from DRF's
defaults.

See DEPR: https://github.com/openedx/edx-platform/issues/33028

KristinAoki committed:
feat: add file size length to returned json (#33068)

bszabo committed:
Revert "fix: refactor MakoService to allow specifying namespace per template (#33061)" (#33070)

This reverts commit d60cdc2305fc7ebb99fbffa5fad535a91f9f0be2.

UsamaSadiq committed:
fix: fix middleware get_response parameter deprecation warning (#33067)

Yagnesh1998 committed:
feat: remove the registration.enable_failure_logging temporary Waffle… (#32977)

* feat: remove the registration.enable_failure_logging temporary WaffleFlag depr84

dyudyunov committed:
Add the default enrollment start date on course creation (#30954)

* fix: main page course listing

The course is visible on the main page right after creation when the feature toggle `CREATE_COURSE_WITH_DEFAULT_ENROLLMENT_START_DATE` is on.

So anonymous users can see them and access the course about page
for the courses without valid data (e.g. they will see the default
course overview)

When courses list filtering is processed it checks the `see_exists`
permission for the anonymous user.
Actually, `see_exists` means `can_load` OR `can_enroll`.

`can_load` is False in our case because the course start in the future.

But `can_enroll` returns True because the course's enrollment_start
and enrollment_end dates are blank:
```
enrollment_start = courselike.enrollment_start or datetime.min.replace(tzinfo=UTC)
enrollment_end = courselike.enrollment_end or datetime.max.replace(tzinfo=UTC)
if enrollment_start < now < enrollment_end:
    debug("Allow: in enrollment period")
    return ACCESS_GRANTED
```

Set the enrollment_start the same as a course start by default

awais786 committed:
chore: adding test to verify default-acl behaviour. (#33074)

* chore: adding test to verify default-acl behaviour.

mphilbrick211 commented:
Hi @dianakhuang! Are you able to enable the tests on this PR again? They bounced back to needing approval.

zainab-amir committed:
feat: add user details to registration response (#33078)

awais786 committed:
Remove matplotlib pin (#32787)

* chore: upgrading matplotlib to latest version.
---------

Authored-by: awais786 <awais786@users.noreply.github.com>

jajjibhai008 committed:
feat: Upgrade Python dependency edx-enterprise

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

moeez96 committed:
refactor: Add logging to mobile_api app (#33064)

mubbsharanwar committed:
fix: update dashboard recommendations endpoint (#33072)

- Update recommendations endpoint from learner recommendations to edx recommendations.
- Added docstring to tell contributors that the recommendations related code has been moved to edx-recommendations plugin.

VAN-1596

ruzniaievdm committed:
feat: add api for help tokens (#33073)

bradenmacdonald committed:
fix: refactor MakoService to allow specifying template more explicitly (Take 2) (#33077)

* fix: refactor MakoService to allow specifying namespace per template (#33061)

* fix: instr. dashboard broken by bulk email reusing HtmlBlock studio_view

* fix: lint issue from unused import

mphilbrick211 commented:
Hi @mehedikhan72! Looks like there's some failing tests - would you mind taking a look?

mehedikhan72 commented:
Hi @mphilbrick211 , on it.

mehedikhan72 commented:
Had to rebase to edit an older commit message, hopefully this should fix it now. There was another failed check and I'm not sure why that was happening. Might need help with that one, once these checks run.

mphilbrick211 commented:
Hi @mehedikhan72 - just checking in on this!

mphilbrick211 commented:
> Hi @mehedikhan72 - just checking in on this!

Hi @mehedikhan72! Friendly ping on this.

mphilbrick211 commented:
Hi @mehedikhan72! Just checking in to see if you plan to pursue this PR?

mehedikhan72 commented:
Hi @mphilbrick211 . I should have responded earlier. I fell sick and got myself occupied into other tasks later on and I think I've lost all the interest to further pursue this PR.

mphilbrick211 commented:
> Hi @mphilbrick211 . I should have responded earlier. I fell sick and got myself occupied into other tasks later on and I think I've lost all the interest to further pursue this PR.

Hi @mehedikhan72 - I'm sorry to hear that. I will close this for now, but we can reopen it if you'd like to work on it again in the future.

openedx-webhooks commented:
@mehedikhan72 Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/32992&entry.1484655318=https://github.com/mehedikhan72&entry.752974735=2023-08-12+00:46&entry.1917517419=2023-10-26+17:45&entry.2133058324=No) so we can improve your experience in the future.

<!-- comment:end_survey -->
<!-- data: eyJqaXJhLXByZS1jbG9zZSI6IG51bGx9 -->
